cmd/kubetoken/validator.go

```diff
@@ -0,0 +1,232 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubetoken
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+var (
+	errEmptyContext = errors.New("context name cannot be empty")
+
+	validationTimeout = 3 * time.Second
+)
+
+type k8sClientProvider interface {
+	Provide(clientApiConfig *clientcmdapi.Config) (kubernetes.Interface, *rest.Config, error)
+}
+
+type preReqCfg struct {
+	ctxName string
+	ns      string
+
+	k8sClientProvider    k8sClientProvider
+	oktetoClientProvider oktetoClientProvider
+}
+
+type option func(*preReqCfg)
+
+func withCtxName(ctxName string) option {
+	return func(cfg *preReqCfg) {
+		cfg.ctxName = ctxName
+	}
+}
+
+func withNamespace(ns string) option {
+	return func(cfg *preReqCfg) {
+		cfg.ns = ns
+	}
+}
+
+func withK8sClientProvider(k8sClientProvider k8sClientProvider) option {
+	return func(cfg *preReqCfg) {
+		cfg.k8sClientProvider = k8sClientProvider
+	}
+}
+
+func withOktetoClientProvider(oktetoClientProvider oktetoClientProvider) option {
+	return func(cfg *preReqCfg) {
+		cfg.oktetoClientProvider = oktetoClientProvider
+	}
+}
+
+func defaultPreReqCfg() *preReqCfg {
+	return &preReqCfg{
+		k8sClientProvider:    okteto.NewK8sClientProvider(),
+		oktetoClientProvider: okteto.NewOktetoClientProvider(),
+	}
+}
+
+// preReqValidator validates that all the pre-reqs to execute the command are met
+type preReqValidator struct {
+	ctxName string
+	ns      string
+
+	k8sClientProvider    k8sClientProvider
+	oktetoClientProvider oktetoClientProvider
+}
+
+// newPreReqValidator returns a new preReqValidator
+func newPreReqValidator(opts ...option) *preReqValidator {
+	cfg := defaultPreReqCfg()
+	for _, opt := range opts {
+		opt(cfg)
+	}
+	return &preReqValidator{
+		ctxName:              cfg.ctxName,
+		ns:                   cfg.ns,
+		k8sClientProvider:    cfg.k8sClientProvider,
+		oktetoClientProvider: cfg.oktetoClientProvider,
+	}
+}
+
+// Validate validates that all the pre-reqs to execute the command are met
+func (v *preReqValidator) Validate(ctx context.Context) error {
+	oktetoLog.Info("validating pre-reqs for kubetoken")
+
+	ctx, cancel := context.WithTimeout(ctx, validationTimeout)
+	defer cancel()
+
+	err := newCtxValidator(v.ctxName, v.k8sClientProvider).Validate(ctx)
+	if err != nil {
+		return fmt.Errorf("invalid context: %w", err)
+	}
+
+	err = newOktetoSupportValidator(ctx, v.ctxName, v.ns, v.k8sClientProvider, v.oktetoClientProvider).Validate(ctx)
+	if err != nil {
+		return fmt.Errorf("invalid okteto support: %w", err)
+	}
+	return nil
+}
+
+// ctxValidator checks that the ctx use to execute the command is an okteto context
+// that has already being added to your okteto context
+type ctxValidator struct {
```

Comment:
NIT: in case of timeout it would be more correct to say something like:

```suggestion
		return fmt.Errorf("context validation failed: %w", err)
```